tests/test_solver.py

```diff
@@ -411,3 +411,25 @@ def test_coupled_block_heart_with_coronaries():
     assert np.isclose(
         np.amin(aortic_pressure), 34.184224686628035, rtol=RTOL_PRES
     )  # min aortic pressure
+
+
+def test_valve_tanh():
+    result = run_test_case_by_name("valve_tanh", output_variable_based=True)
+    upstream_pressure = result[result.name == "pressure:upstream_vessel:valve"][
```

Comment:
Yup, the connections are based on whatever the block name is. In the test case, the upstream and downstream blocks are vessels and I just named them `upstream_vessel` and `downstream_vessel`, but they don't even need to be vessels. They could be any block. The main use case will likely be having a valve downstream of a heart chamber block (which we are also working on and will have its own pull request soon), but in general it could be any block.